modules/pipeline/pipengine/actionexecutor/logic/status.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logic
+
+import "github.com/erda-project/erda/apistructs"
+
+func TransferStatus(status string) apistructs.PipelineStatus {
+	switch status {
+
+	case string(apistructs.StatusError):
+		return apistructs.PipelineStatusError
+
+	case string(apistructs.StatusUnknown):
+		return apistructs.PipelineStatusUnknown
+
+	case string(apistructs.StatusCreated):
+		return apistructs.PipelineStatusCreated
+
+	case string(apistructs.StatusUnschedulable), "INITIAL":
+		return apistructs.PipelineStatusQueue
+
+	case string(apistructs.StatusRunning), "ACTIVE":
+		return apistructs.PipelineStatusRunning
+
+	case string(apistructs.StatusStoppedOnOK), string(apistructs.StatusFinished):
+		return apistructs.PipelineStatusSuccess
+
+	case string(apistructs.StatusStoppedOnFailed), string(apistructs.StatusFailed):
+		return apistructs.PipelineStatusFailed
+
+	case string(apistructs.StatusStoppedByKilled):
+		return apistructs.PipelineStatusStopByUser
+
+	case string(apistructs.StatusNotFoundInCluster):
+		// scheduler 返回 job 在 cluster 中不存在 (在 etcd 中存在)，对应为 启动错误
+		// 典型场景：created 成功，env key 为数字，导致 start job 时真正去创建 k8s job 时失败，即启动失败
```

Comment:
use English is better.

Reply:
and rethink this situation, is right or not.

Reply:
used english, maybe notfound in cluster status shouldn't be start error